src/LocaleService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\argo;
+
+use Drupal\locale\StringDatabaseStorage;
+use Drupal\tableau_i18n_locale\TableauLocaleStringInterface;
+
+/**
+ * Handles locale export and translation.
+ */
+class LocaleService {
+
+  /**
+   * Creates a new instance of the Argo locale translation service.
+   */
+  public function __construct(StringDatabaseStorage $stringStorage) {
+    $this->stringStorage = $stringStorage;
+  }
+
+  /**
+   * Export a list of UI strings for translation.
+   *
+   * @param string $langcode
+   *   The langcode for which to export UI strings.
+   * @param array $options
+   *   Options to control the export.
+   *   [
+   *     'include_translations' => 'Whether or not to include already translated
+   *       values',
+   *   ]
+   *
+   * @return array
+   *   List of UI strings:
+   *    [
+   *     'string' => 'The original string value in the source language',
+   *     'translation' => 'A copy of the original string to be replaced by the
+   *       translator',
+   *     'context' => 'Context for the string.',
+   *     'url' => 'The url at which the string is found.',
+   *   ]
+   */
+  public function export(string $langcode, array $options = []) {
+    // Merge in default options.
+    $options += ['include_translations' => FALSE];
+    $items = [];
+    // @todo remove dependency on tableau_i18n_locale module. Maybe invoke an event/hook so that custom Tableau modules
+    //   can hook into this?
+    $strings = $this->stringStorage->getStrings([
+      'translated' => $options['include_translations'],
+      'visibility' => TableauLocaleStringInterface::EXTERNAL,
+    ]);
+
+    foreach ($strings as $string) {
+      // Try and retrieve metadata about the origins of this string.
+      $locations = $string->getLocations();
+      $url = '';
+      if (isset($locations['path'])) {
+        $url = implode(',', array_keys($locations['path']));
```

Comment:
Would this run into any risk of being too long for storage? Do we need to consider truncating it?

Reply:
This is purely on export, the import part is handled by Drupal core and we don't have to worry about it